packages/protocol/test/governance/epochrewards.ts

```diff
@@ -0,0 +1,576 @@
+      assertEqualBN(target, targetVotingYieldParams.initial)
+      assertEqualBN(max, targetVotingYieldParams.max)
+      assertEqualBN(adjustmentFactor, targetVotingYieldParams.adjustmentFactor)
+    })
+
+    it('should have set the rewards multiplier parameters', async () => {
+      const [max, underspend, overspend] = await epochRewards.getRewardsMultiplierParameters()
+      assertEqualBN(max, rewardsMultiplier.max)
+      assertEqualBN(underspend, rewardsMultiplier.adjustments.underspend)
+      assertEqualBN(overspend, rewardsMultiplier.adjustments.overspend)
+    })
+
+    it('should not be callable again', async () => {
+      await assertRevert(
+        epochRewards.initialize(
+          registry.address,
+          targetVotingYieldParams.initial,
+          targetVotingYieldParams.max,
+          targetVotingYieldParams.adjustmentFactor,
+          rewardsMultiplier.max,
+          rewardsMultiplier.adjustments.underspend,
+          rewardsMultiplier.adjustments.overspend,
+          targetVotingGoldFraction,
+          targetValidatorEpochPayment
+        )
+      )
+    })
+  })
+
+  describe('#setTargetVotingGoldFraction()', () => {
+    describe('when the fraction is different', () => {
+      const newFraction = targetVotingGoldFraction.plus(1)
+
+      describe('when called by the owner', () => {
+        let resp: any
+
+        beforeEach(async () => {
+          resp = await epochRewards.setTargetVotingGoldFraction(newFraction)
+        })
+
+        it('should set the target voting gold fraction', async () => {
+          assertEqualBN(await epochRewards.getTargetVotingGoldFraction(), newFraction)
+        })
+
+        it('should emit the TargetVotingGoldFractionSet event', async () => {
+          assert.equal(resp.logs.length, 1)
+          const log = resp.logs[0]
+          assertContainSubset(log, {
+            event: 'TargetVotingGoldFractionSet',
+            args: {
+              fraction: newFraction,
+            },
+          })
+        })
+
+        describe('when called by a non-owner', () => {
+          it('should revert', async () => {
+            await assertRevert(
+              epochRewards.setTargetVotingGoldFraction(newFraction, {
+                from: nonOwner,
+              })
+            )
+          })
+        })
+      })
+
+      describe('when the fraction is the same', () => {
+        it('should revert', async () => {
+          await assertRevert(epochRewards.setTargetVotingGoldFraction(targetVotingGoldFraction))
+        })
+      })
+    })
+  })
+
+  describe('#setTargetValidatorEpochPayment()', () => {
+    describe('when the payment is different', () => {
+      const newPayment = targetValidatorEpochPayment.plus(1)
+
+      describe('when called by the owner', () => {
+        let resp: any
+
+        beforeEach(async () => {
+          resp = await epochRewards.setTargetValidatorEpochPayment(newPayment)
```

Comment:
Nit: I think it's conventional, when unit testing a particular method, to have the call be in an `it`, even if you could deduplicate it in a `beforeEach`. `beforeEach` is more about the setup needed before the call is made.